docs/deployment.md

```diff
@@ -0,0 +1,485 @@
+KubeVirt Deployment
+=
+
+# Preface
+When deploying [KubeVirt](https://kubevirt.io/) onto a [Kubernetes](https://kubernetes.io/) cluster, we deploy the following:
+
+- `virt-operator` and its dependencies
+- KubeVirt CustomResource
+
+This is done using the [kubectl](https://kubernetes.io/docs/reference/kubectl/) command line tool or programmatically.
+
+```shell
+# Point at latest stable release
+$ export RELEASE=$(curl https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)
+# Deploy the virt-operator
+$kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-operator.yaml
+# Create the KubeVirt CR (instance deployment request) which triggers the actual installation
+$kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-cr.yaml
+# Wait until all KubeVirt components are up
+$kubectl --namespace=kubevirt wait kubevirt kubevirt --for condition=Available
+```
+
+# Virt Operator Deployment
+
+`virt-operator` is the component responsible for deploying the components of the KubeVirt project.
+
+## Namespace
+A dedicated [Namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) is created for KubeVirt's components.
+
+```shell
+kubectl get namespace kubevirt -o yaml
+```
+
+## KubeVirt CustomResourceDefinition
+KubeVirt extends Kubernetes using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs).
+
+The first deployed CRD is `kubevirts.kubevirt.io` which allows us to later create the `KubeVirt` CustomResource.
```

Comment:
Hi Orel, this is great idea! I think it would be longer than usual to read, maybe you can make table instead? Similar to what we have in SSP docs [1].

[1] https://github.com/kubevirt/ssp-operator/blob/main/docs/installation.md#kubernetes

Reply:
I thought about it in the beginning, and came to the conclusion that the `kubectl get` commands will break.

Reading the document from start to finish is indeed not trivial, but using an IDE (or a similar tool that helps read markdown files) is more convenient - since it provides a generated table of content.

Reply:
Additionally, there is a feature in GitHub to view the TOC automatically:
https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

Unfortunately it only works for merged files.

Reply:
Okay. What about this suggestion to make it shorter:

## Export Proxy Resources

Not deployed by default. Deployed only when `VMExport` feature gate is added to KubeVirt CR.

```shell
# Cluster Resources
kubectl get clusterroles.rbac.authorization.k8s.io kubevirt-exportproxy -o yaml
kubectl get clusterrolebindings.rbac.authorization.k8s.io kubevirt-exportproxy -o yaml

# Namespace Resources
kubectl get --namespace=kubevirt serviceaccount kubevirt-exportproxy -o yaml
kubectl get --namespace=kubevirt roles.rbac.authorization.k8s.io  kubevirt-exportproxy -o yaml
kubectl get --namespace=kubevirt rolebindings.rbac.authorization.k8s.io kubevirt-exportproxy -o yaml
kubectl get --namespace=kubevirt secret kubevirt-exportproxy-certs -o yaml
kubectl get --namespace=kubevirt deployments.apps virt-exportproxy -o yaml
kubectl get --namespace=kubevirt service virt-exportproxy -o yaml
kubectl get --namespace=kubevirt configmap kubevirt-ca -o yaml
kubectl get --namespace=kubevirt configmap kubevirt-export-ca -o yaml
kubectl get --namespace=kubevirt secret kubevirt-ca -o yaml
kubectl get --namespace=kubevirt secret kubevirt-export-ca -o yaml
```

Reply:
Interesting suggestion, I'll consider it.
The issue with it is that its harder to insert explanations between the list of resources.